fineract-provider/src/test/java/org/apache/fineract/infrastructure/core/LiquibaseStepDefinitions.java

```diff
@@ -99,9 +106,18 @@ public LiquibaseStepDefinitions() {
 
         Then("The database migration did not do anything", () -> {
             verify(databaseStateVerifier).isLiquibaseDisabled();
+            verify(fineractProperties).getMode();
             verifyNoMoreInteractions(databaseStateVerifier);
-            verifyNoInteractions(tenantDetailsService, tenantStoreDataSource, fineractProperties, liquibaseFactory,
-                    tenantDataSourceFactory);
+            verifyNoMoreInteractions(fineractProperties);
+            verifyNoInteractions(tenantDetailsService, tenantStoreDataSource, liquibaseFactory, tenantDataSourceFactory);
+        });
+
+        Then("The database migration did not do anything, because it is not a write instance", () -> {
```

Comment:
Just a note: since this is related to liquibase, you can leave it here instead of moving this into the other class I mentioned above.